### PR TITLE
chore: Update azure.Docker file to work with Azure DevOps Pipelines

### LINKED ISSUE #2502: Cannot use Garden CI Container in Azure DevOps
### Current Behavior

I am trying to set up a CI/CD Pipeline for my Garden deployment on Azure DevOps Pipelines, this fails with the current Garden provided image as Azure DevOps works by adding a user to the container to run the tasks via.

### Reproducible example

A sample yaml pipeline

```
- job: gardenBuild
  condition: true
  container:
    image: gardendev/garden-azure:0.12.23-0
  steps:
  - script: garden test --logger-type=basic --env=systest
```

Gives the following output when ran

```
Try to create a user with UID '1001' inside the container.
/usr/bin/docker exec  1bc0e033d550c24855c38ec627310834220dbf480062f5b3d3b4a77700ba2892 bash -c "getent passwd 1001 | cut -d: -f1 "
/usr/bin/docker exec  1bc0e033d550c24855c38ec627310834220dbf480062f5b3d3b4a77700ba2892 useradd -m -u 1001 vsts_azpcontainer
OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "useradd": executable file not found in $PATH: unknown
```
### Workaround

I am working around this by creating my own image which needs to be updated on Garden releases:

```
FROM gardendev/garden-azure:0.12.22

RUN apk add --no-cache --virtual .pipeline-deps readline linux-pam  \
  && apk add bash sudo shadow \
  && apk del .pipeline-deps

LABEL "com.azure.dev.pipelines.agent.handler.node.path"="/usr/local/bin/node"

ENTRYPOINT []
```

### Suggested solution(s)

The fix is adding the useradd binary to the image so that it works with Azure DevOps. 

I understand that you may not want to as it increases the surface area of the container.

If it can be fixed great, if not - it is probably worth adding a note to your FAQ section about this for any future Azure DevOps users.

### Your environment

Azure DevOps
Azure Kubernetes Service

### PR DESCRIPTION
**Pull request by ITHedgeHog**
**What this PR does / why we need it**:

Adds the correct items needed for this Docker image to run on Azure DevOps

**Which issue(s) this PR fixes**:
Fixes #2502 

**Special notes for your reviewer**:

@edvald I checked the code I copied from Stack Overflow, it was in fact all needed -  I purposely chose to add the extra bits for Azure DevOps in a second run command so that it creates a separate layer for reuse in other images as necessary. I'm pretty sure this can all be collapsed in to a single run command if you prefer.

The Endpoint and label are both required by Azure DevOps for the image to run.

I've deployed this image to our internal registry and updated our CI/CD pipeline to use it.

### CODE REVIEW

**Review by edvald — CHANGES_REQUESTED** [review]
Thanks @ITHedgeHog! Maybe being a bit nitpicky, but I had a couple of small comments, should be quick to address .)

File: support/azure.Dockerfile [hunk 1]
```diff
@@ -1,11 +1,29 @@
 ARG TAG=latest
 FROM gardendev/garden:${TAG}
+ENV KUBELOGIN_VERSION="v0.0.9"
```
[the remark this answers is not in the document]
Comment (ITHedgeHog) [reply support/azure.Dockerfile]: I've moved this.

File: support/azure.Dockerfile [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+  && apk del .pipeline-deps
+
+# Ensure kubelogin is available
+RUN wget -O kubelogin-linux-amd64.zip https://github.com/Azure/kubelogin/releases/download/$KUBELOGIN_VERSION/kubelogin-linux-amd64.zip  \
+ && unzip kubelogin-linux-amd64.zip \
+ && cp bin/linux_amd64/kubelogin /usr/local/bin/ \
+ && rm kubelogin-linux-amd64.zip
+
+
+LABEL "com.azure.dev.pipelines.agent.handler.node.path"="/usr/local/bin/node"
```
[the remark this answers is not in the document]
Comment (ITHedgeHog) [reply support/azure.Dockerfile:22]: Comment added

File: support/azure.Dockerfile [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
-  && apk add python3-dev && pip3 install -U pip \
+  && apk add  python3-dev && pip3 install -U pip \
 # Actual azure cli
   && pip3 --no-cache-dir install azure-cli \
 # Remove build dependencies
   && apk del --purge build
+
+# Azure DevOps Pipeline Dependencies
+RUN apk add --no-cache --virtual .pipeline-deps readline linux-pam  \
+  && apk add bash sudo shadow \
```
[the remark this answers is not in the document]
Comment (ITHedgeHog) [reply support/azure.Dockerfile]: I've merged the two run statements. I tried building it without adding readline and linux-pam however without those I could not add bash sudo or shadow.

**Review by edvald — APPROVED** [review]

**Pull request merged**